src/researches/getPassiveVoice.js

```diff
@@ -1,110 +1,122 @@
-var getSentences = require( "../stringProcessing/getSentences.js" );
-var stripHTMLTags = require( "../stringProcessing/stripHTMLTags.js" ).stripFullTags;
-var getLanguage = require( "../helpers/getLanguage.js" );
-var Sentence = require( "../values/Sentence.js" );
+const getSentences = require( "../stringProcessing/getSentences.js" );
+const stripHTMLTags = require( "../stringProcessing/stripHTMLTags.js" ).stripFullTags;
+const getLanguage = require( "../helpers/getLanguage.js" );
+const Sentence = require( "../values/Sentence.js" );
 
-// Imports used for English, French and Spanish.
-var getSentencePartsDefault = require( "./passiveVoice/getSentenceParts.js" );
-var determinePassivesDefault = require( "./passiveVoice/determinePassives" );
+const forEach = require( "lodash/forEach" );
 
-// Imports used for German.
-var getSentencePartsGerman = require( "./german/passiveVoice/getSentenceParts.js" );
-var determinePassivesGerman = require( "./german/passiveVoice/determinePassives.js" );
+const determinePassiveSentencePart = require( "./passiveVoice/periphrastic/determinePassiveSentencePart.js" );
+const determinePassiveSentence = require( "./passiveVoice/morphological/determinePassiveSentence.js" );
 
-var forEach = require( "lodash/forEach" );
+const getSentencePartsDefault = require( "./passiveVoice/periphrastic/getSentenceParts.js" );
+const getSentencePartsGerman = require( "./german/passiveVoice/getSentenceParts.js" );
+
+const morphologicalLanguages = [ "ru", "tr" ];
+const periphrasticLanguages = [ "en", "de", "nl", "fr", "es", "it", "pt", "cn" ];
+const morphologicalAndPeriphrasticLanguages = [ "sv", "da", "nb" ];
 
 /**
- * Gets the sentence parts from a sentence by determining sentence breakers.
+ * Looks for morphological passive voice.
  *
- * @param {string} sentence The sentence to split up in sentence parts.
- * @param {string} language The language to use for determining how to get sentence parts.
- * @returns {Array} The array with all parts of a sentence that have an auxiliary.
+ * @param {Array} sentenceObjects Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {object} The number of passives found in the text and the passive sentences.
  */
-var getSentenceParts = function( sentence, language ) {
-	var sentenceParts = [];
-
-	switch( language ) {
-		case "de":
-			sentenceParts = getSentencePartsGerman( sentence );
-			break;
-		case "fr":
-			sentenceParts = getSentencePartsDefault( sentence, "fr" );
-			break;
-		case "es":
-			sentenceParts = getSentencePartsDefault( sentence, "es" );
-			break;
-		case "en":
-		default:
-			sentenceParts = getSentencePartsDefault( sentence, "en" );
-			break;
-	}
-	return sentenceParts;
+let getPassivesMorphological = function( sentenceObjects, language ) {
```

Comment:
I think the name flows better if it would be called `getMorphologicalPassives`. I am also wondering if the `Passives` is necessary here. Maybe `getMorphologicalSentences` is better? Or does that not make sense language-wise? What do you think?

Reply:
'Morphological sentences' doesn't make sense as a concept for me. Every sentence consists of morphemes. Morphemes are the smallest meaning-bearing things in a language: https://sites.google.com/a/sheffield.ac.uk/aal2013/branches/morphology/what-is-morphology

Reply:
I agree with Irene, `getMorphologicalSentences` sounds really awkward to my ears.